kong/resty/dns_client/init.lua

```diff
@@ -0,0 +1,782 @@
+local cjson = require("cjson.safe")
+local utils = require("kong.resty.dns_client.utils")
+local tablex = require("pl.tablex")
+local mlcache = require("kong.resty.mlcache")
+local resolver = require("resty.dns.resolver")
+
+local now       = ngx.now
+local log       = ngx.log
+local ERR       = ngx.ERR
+local WARN      = ngx.WARN
+local DEBUG     = ngx.DEBUG
+local ALERT     = ngx.ALERT
+local timer_at  = ngx.timer.at
+
+local type          = type
+local pairs         = pairs
+local ipairs        = ipairs
+local math_min      = math.min
+local string_lower  = string.lower
+local table_insert  = table.insert
+
+local parse_hosts   = utils.parse_hosts
+local ipv6_bracket  = utils.ipv6_bracket
+local search_names  = utils.search_names
+local get_next_round_robin_answers     = utils.get_next_round_robin_answers
+local get_weighted_round_robin_answers = utils.get_weighted_round_robin_answers
+
+local req_dyn_hook_run_hooks = require("kong.dynamic_hook").run_hooks
+local cycle_aware_deep_copy  = require("kong.tools.utils").cycle_aware_deep_copy
+
+
+-- Constants and default values
+
+local DEFAULT_ERROR_TTL   = 1     -- unit: second
+local DEFAULT_STALE_TTL   = 4
+local DEFAULT_EMPTY_TTL   = 30
+-- long-lasting TTL of 10 years for hosts or static IP addresses in cache settings
+local LONG_LASTING_TTL    = 10 * 365 * 24 * 60 * 60
+
+local DEFAULT_ORDER = { "LAST", "SRV", "A", "AAAA", "CNAME" }
+
+local TYPE_SRV      = resolver.TYPE_SRV
+local TYPE_A        = resolver.TYPE_A
+local TYPE_AAAA     = resolver.TYPE_AAAA
+local TYPE_CNAME    = resolver.TYPE_CNAME
+local TYPE_LAST     = -1
+
+local NAME_TO_TYPE = {
+  SRV     = TYPE_SRV,
+  A       = TYPE_A,
+  AAAA    = TYPE_AAAA,
+  CNAME   = TYPE_CNAME,
+  LAST    = TYPE_LAST,
+}
+
+local TYPE_TO_NAME = {
+  [TYPE_SRV]      = "SRV",
+  [TYPE_A]        = "A",
+  [TYPE_AAAA]     = "AAAA",
+  [TYPE_CNAME]    = "CNAME",
+}
```

Comment:
We should add a simple comment to explain why there is no `TYPE_LAST=??`.

Reply:
or we can just add it, to make it symmetrical.